src/middlewared/middlewared/plugins/pool.py

```diff
@@ -2450,6 +2450,13 @@ def unlock(self, job, id, options):
 
         Uploading a json file which contains encrypted dataset keys can be specified with
         `unlock_options.key_file`. The format is similar to that used for exporting encrypted dataset keys.
+
+        `attachments_list_mode` and `attachments_list` controls which attachments (shares, tasks)
+        should be put in action after unlocking dataset(s). By default all attachments are handled, which might
+        theoretically lead to service interruption when daemons configurations are reloaded (this should not happen,
+        and if this happens it should be considered a bug). As TrueNAS does not have a state for resources that
```

Comment:
Forgive me, I've read this a few times and I still don't quite understand what this is explaining. I'm not really grok'ing the title of this PR and this comment just further confuses me. Specifically the bit about `put the system into an inconsistent state so it should really never be used.`

If we feel this strongly about it, then why do we even have this option?

Reply:
By "inconsistent state" I mean samba shares or cloud sync tasks or whatever that were LOCKED because underlying dataset was locked. Then, if underlying services were not reloaded when unlocking the dataset, the shares or tasks will end up being shown as non-locked, but, nevertheless, non-functional, that's what original ticket was about. There is no reason not to reload anything, because scst, NFS and Samba all can add shares without service interruption. We only use this option for failover to avoid double service reload.

Reply:
Okay thanks, so am I understanding correctly that `ALLOW` actually means ignore the reloading/restarting of services and `DENY` means reload/restart?

Reply:
Okay, I understand now. You've added an allow/deny list (I think...). Anyways, I do not like the verbiage change here. The previous API verbiage made more sense to me.

With your changes, we only have 2 options in the `enum=`. Why do we not simplify this to a singular boolean? Something like `restart_or_reload_services` as a boolean and default it to `True` and on failover we set it to `False`?